Data/Primitive/Types.hs

```diff
@@ -205,3 +211,45 @@ derivePrim(Ptr a, Ptr, sIZEOF_PTR, aLIGNMENT_PTR,
 derivePrim(FunPtr a, FunPtr, sIZEOF_PTR, aLIGNMENT_PTR,
            indexAddrArray#, readAddrArray#, writeAddrArray#, setAddrArray#,
            indexAddrOffAddr#, readAddrOffAddr#, writeAddrOffAddr#, setAddrOffAddr#)
+
+-- Prim instances for newtypes in Foreign.C.Types
+deriving instance Prim CChar
+deriving instance Prim CSChar
+deriving instance Prim CUChar
+deriving instance Prim CShort
+deriving instance Prim CUShort
+deriving instance Prim CInt
+deriving instance Prim CUInt
+deriving instance Prim CLong
+deriving instance Prim CULong
+deriving instance Prim CPtrdiff
+deriving instance Prim CSize
+deriving instance Prim CWchar
+deriving instance Prim CSigAtomic
+deriving instance Prim CLLong
+deriving instance Prim CULLong
+#if MIN_VERSION_base(4,10,0)
+deriving instance Prim CBool
+#endif
+deriving instance Prim CIntPtr
+deriving instance Prim CUIntPtr
+deriving instance Prim CIntMax
+deriving instance Prim CUIntMax
+deriving instance Prim CClock
+deriving instance Prim CTime
+deriving instance Prim CUSeconds
+deriving instance Prim CSUSeconds
+deriving instance Prim CFloat
+deriving instance Prim CDouble
+
+-- Prim instances for newtypes in System.Posix.Types
+deriving instance Prim CDev
+deriving instance Prim CIno
+deriving instance Prim CMode
+deriving instance Prim COff
+deriving instance Prim CPid
+deriving instance Prim CSsize
+#if MIN_VERSION_base(4,10,0)
+deriving instance Prim CClockId
+#endif
+deriving instance Prim Fd
```

Comment:
This surely isn't all of the newtypes found in `System.Posix.Types`, right? I'm looking at http://hackage.haskell.org/package/base-4.11.0.0/docs/System-Posix-Types.html, and there's a good deal more of them than listed here.

Also, not all of these types are exposed on all operating systems. You'll need to guard each one with CPP as shown in http://hackage.haskell.org/package/base-4.11.0.0/docs/src/System.Posix.Types.html. For instance:

```haskell
#if defined(HTYPE_DEV_T)
deriving instance Prim CDev
#endif
...
```